packages/react-router/lib/dom/ssr/browser.tsx

```diff
@@ -179,6 +219,24 @@ function createHydratedRouter(): RemixRouter {
   });
   ssrInfo.router = router;
 
+  // Do this after creating the router so ID's have been added to the routes
+  // that we can use as keys in the manifest. `router.routes` will contain the
+  // route IDs, `props.routes` may not.
+  if (propRoutes && ssrInfo) {
+    if (propRoutesError) {
+      console.warn(propRoutesError);
+    } else {
+      let rootDataRoute = router.routes[0];
+      for (let route of rootDataRoute.children || []) {
+        addPropRoutesToRemix(
+          ssrInfo,
+          route as DataRouteObject,
+          rootDataRoute.id
+        );
```

Comment:
Add the new routes to the Remix `manifest`/`routeModules` structures as if they came from the server - they just won't have a `route.module`